docs/federation.md

```diff
@@ -0,0 +1,116 @@
+# Federation
+
+## Introduction
+
+The DACCS Network (“Network”) is a federated network that consists of independently operated and maintained computing
+infrastructure (“Node”) that deploy the DACCS software stack and connect with one another under the framework laid out
+in this document. While anyone is free is deploy the software stack on their own hardware and use it as they choose, in
+order to join the Network as a Node, the Node needs to provide certain capabilities and minimum operating standards, as
+laid out in this document, and which must be agreed upon by the person(s) responsible for maintaining the Node (“Node
+Administrator”).
+
+## Principles of Federation
+
+The following sections describe the requirements for operating and maintaining a Node.
+
+### Operations
+
+- Deployment of software stack: Each Node should deploy the DACCS software stack. Node Administrators are required to
+  upgrade the software stack to the latest DACCS stack within two weeks of the updates release date.
+- Node maintenance: Node Administrators accept that joining the Network implies that they are responsible for
+  provisioning, funding and maintaining their own Node.
+- Node security: Node Administrators are responsible for maintaining secure access to their node and for coordinating
+  with
+  the hosting institution regarding any security related considerations prior to the deployment of the Node. Admin
+  access
+  to the Node must be restricted to listed Node Administrators.
+- Downtime policy: A Node should strive to maintain downtime of no more than 5%. If a Node needs to be taken offline for
+  scheduled maintenance purposes, the Node Administrator must provide users at least 2-weeks notice.
+- New user signups: Node Administrators are responsible for the signups of users on their own Nodes. All new signup
+  requests should be verified and approved or rejected within 2 business days.
```

Comment:
This 2 business days can be a bit short, especially in a high season vacation.

Reply:
What would you recommend?

Reply:
Maybe add the word "ideally" in front of the "2 business days"?

I like us to convey the idea that we will try hard to meet the level of service but it's not a warranty.  I fear this document will induce users to think this is our TOS (terms of service) as in a commercial contract.  I don't think we are at that level yet, especially we are all non profit organisations, providing the service for free to our users.

Same spirit for all other limits discussed in this document.

Reply:
There are two goals with this document: 

1. providing Node Administrators with a set of guidelines that we _recommend_ they follow. If they choose not to follow these, then _their node_ may not run as well as we'd like but that's ok.
2. providing Node Administrators with a set of guidelines that they _must_ follow. If they choose not to follow these, then the node will degrade the quality of the network itself. If they do not follow these then we may choose to remove their node from the network.

This is making me think that we should reorganize this document to clarify these two goals.

Reply:
Agreed. In the current state, I had the wrong impression all the guidelines are "must follow".